### PR TITLE
ci: Write tests using Go's testing package

### PR DESCRIPTION
**Pull request by kklin**

### CODE REVIEW

File: quilt-tester/tests/10-network/network_test.go [hunk 1]
```diff
@@ -54,61 +62,43 @@ func (ct commandTime) String() string {
...
-				failed = true
-				fmt.Println(
-					".. FAILED, could ping unauthorized containers")
-				for _, unauthorized := range res.pingUnauthorized {
-					fmt.Printf(".... %s\n", unauthorized)
-				}
+				t.Errorf("could ping unauthorized containers from %v:\n%s",
+					res.container, testFailureSlice(res.pingUnauthorized))
 			}
 			if len(res.pingUnreachable) != 0 {
```
Comment (ejj) [inline quilt-tester/tests/10-network/network_test.go]: Have you considered using the assert library?  (Totally fine to do it in a different patch, or never)

Comment (kklin) [inline quilt-tester/tests/10-network/network_test.go]: Yeah, I'm sure it could be helpful when it comes up, but I don't feel a need to transition the current tests to assert.

File: quilt-tester/tests/inbound-public/inbound_public_test.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package main
```
Comment (ejj) [inline quilt-tester/tests/inbound-public/inbound_public_test.go]: Is this a new test?  Separate patch?

Comment (ejj) [inline quilt-tester/tests/inbound-public/inbound_public_test.go]: Ah nvm  Just moved code and a confused differ.
